### PR TITLE
feat: Device Registration vue change

### PR DESCRIPTION
**Pull request by talwinder50**
Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: pkg/restapi/device/operations.go [hunk 1]
```diff
@@ -24,16 +25,16 @@ import (
 
 // Endpoints.
 const (
-	registerBeginPath  = "register/begin"
-	registerFinishPath = "register/finish"
-	loginBeginPath     = "login/begin"
-	loginFinishPath    = "login/finish"
+	registerBeginPath  = "/register/begin"
```
Comment (llorllale) [inline pkg/restapi/device/operations.go:28]: [`deviceBasePath`](https://github.com/trustbloc/edge-agent/blob/ffd33ea110d37b9d129340d1420bd5e2496553f4/cmd/http-server/startcmd/start.go#L152) already has a trailing slash... no need to add it here

File: pkg/restapi/device/operations.go [hunk 2]
```diff
@@ -296,11 +297,15 @@ func (o *Operation) getUserData(w http.ResponseWriter, r *http.Request) (userDat
...
 		common.WriteErrorResponsef(w, logger,
-			http.StatusInternalServerError, "failed to read user session cookie: %s", err.Error())
+			http.StatusInternalServerError, "failed to get user data %s:", err.Error())
 
 		return nil, false
 	}
 
+	displayName := strings.Split(fmt.Sprintf("%v", userSub), "@")[0]
+
+	userData.FamilyName = displayName
```
Comment (llorllale) [inline pkg/restapi/device/operations.go:307]: FamilyName == displayName ?

Comment (talwinder50) [reply pkg/restapi/device/operations.go:307]: we had to pick one from our existing user data stuct (FamilyName, Name, Sub, Email and GivenName ) equivalent to device display name. 
func NewDevice(usr *user.User) *Device {
	device := &Device{}
	device.ID = usr.Sub
	device.Name = usr.Name
	device.DisplayName = usr.FamilyName
	device.Credentials = []webauthn.Credential{}

	return device
}

Yes right now FamilyName == displayName

File: pkg/restapi/device/operations.go (hunk 1 above)
Comment (talwinder50) [reply pkg/restapi/device/operations.go:28]: not adding trailing slash gives 404 page not found error. I noticed even 	oidcLoginPath    = "/login" starts with training slash. It doesnt work without that

Comment (llorllale) [reply pkg/restapi/device/operations.go:28]: ok

**Review by llorllale — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**